From e42a0566079a86c3b637b058ac6a5ad3b3ee6f5b Mon Sep 17 00:00:00 2001
From: Eric Ruei <e-ruei1@ti.com>
Date: Thu, 25 May 2017 16:44:51 -0400
Subject: [PATCH 1/3] qtwebengine: fix the system crash due to mismatched GLES2
  library version

- define QT_LIB_GLES2 as "libGLESv2.so.1"
- remove the Q_UNREACHABLE() which causes unnecessary system crash at
  GLSurface::CreateOffscreenGLSurface(

Upstream-Status: Inappropriate
This patch is for TI only because the libGLESv2.so.1 is provided by IMG DDK 1.14.
The issue is expected to be resolved by DDK 1.15 or DDK 1.16 when MESA-style
libraries are provided.

Signed-off-by: Eric Ruei <e-ruei1@ti.com>
---
 src/core/gl_surface_qt.cpp      |  5 +++--
 src/core/surface_factory_qt.cpp | 11 +++++++++--
 2 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/src/core/gl_surface_qt.cpp b/src/core/gl_surface_qt.cpp
index e7b4536..f020be9 100644
--- a/src/core/gl_surface_qt.cpp
+++ b/src/core/gl_surface_qt.cpp
@@ -619,8 +619,9 @@ GLSurface::CreateOffscreenGLSurface(const gfx::Size& size)
     default:
         break;
     }
-    LOG(ERROR) << "Requested OpenGL platform is not supported.";
-    Q_UNREACHABLE();
+    LOG(ERROR) << "Requested OpenGL platform is not supported.";
+    // This is no longer an unreachable code. It is OK to return NULL if any prior operation fails
+    // Q_UNREACHABLE();
     return NULL;
 }
 
diff --git a/src/core/surface_factory_qt.cpp b/src/core/surface_factory_qt.cpp
index 48c91bf..134a866 100644
--- a/src/core/surface_factory_qt.cpp
+++ b/src/core/surface_factory_qt.cpp
@@ -57,8 +57,13 @@
 #endif
 #ifndef QT_LIBDIR_GLES2
 #define QT_LIBDIR_GLES2 QT_LIBDIR_EGL
+#endif
+
+#ifndef QT_LIB_GLES2
+#define QT_LIB_GLES2 "libGLESv2.so.1"
 #endif
 
+
 namespace QtWebEngineCore {
 
 base::NativeLibrary LoadLibrary(const base::FilePath& filename) {
@@ -79,8 +84,10 @@ bool SurfaceFactoryQt::LoadEGLGLES2Bindings(AddGLLibraryCallback add_gl_library,
     if (!eglLibrary)
         return false;
 
-    base::FilePath libGLES2Path = QtWebEngineCore::toFilePath(QT_LIBDIR_GLES2);
-    libGLES2Path = libGLES2Path.Append("libGLESv2.so.2");
+    base::FilePath libGLES2Path = QtWebEngineCore::toFilePath(QT_LIBDIR_GLES2);
+    // It does not make sence to expect the version of libGLESv2 to be 2 as libGLESv2.so.2 
+    // It will be better to use another #define for user to reconfigure the library name
+    libGLES2Path = libGLES2Path.Append(QT_LIB_GLES2);
     base::NativeLibrary gles2Library = LoadLibrary(libGLES2Path);
     if (!gles2Library)
         return false;
-- 
1.9.1

